docs/configmap.md

```diff
@@ -0,0 +1,109 @@
+# Config Map
+
+## About
+The `configmap` which used to tune the `managed-upgrade-operator`. It has various configurable values.
+
+## How to use it
+
+### For OSD:
+The `configmap` will be synced to the clusters from the hive cluster via selectorsyncset. You can update the `configmap` via [`managed-cluster-config`](https://github.com/openshift/managed-cluster-config/tree/master/deploy/managed-upgrade-operator-config).
+
+### For ARO:
+TBD
+
+
+## Configurable knobs
+#### configManager
+
+Please refer to the doc [`configmanager`](../configmanager.md)
+
+#### maintenance
+
+| Key | Description |
+| --- | --- |
+| controlPlaneTime | maintenance window created in alertmanager for controlplane upgrade in minutes including all the low and medium alerts, default is 90 |
+| ignoredAlerts | a list of particular alerts to silent which is not covered by the default maintenance |
+| controlPlaneCriticals | critial alerts might fire during controlplane upgrade |
+
+Example:
+```
+    maintenance:
+      controlPlaneTime: 90
+      ignoredAlerts:
+        controlPlaneCriticals:
+        - ClusterOperatorDown
+        - ClusterOperatorDegraded
+```
+
+#### scale
+
+| Key | Description |
+| --- | --- |
+| timeOut | timeout window for the extra workload scale up in minutes, default is 30 |
+
+Example:
+```
+    scale:
+      timeOut: 30
+```
+
+#### upgradeWindow
+
+| Key | Description |
+| --- | --- |
+| delayTrigger | a time window to indicate the the cluster upgrade is deleyed against the schedule in minutes, default is 30 |
+| timeOut | a time window to indicate the whole upgrade process should be finished within in minutes, default is 120 |
+
+Example:
+```
+    upgradeWindow:
+      delayTrigger: 30
+      timeOut: 120
+```
+
+#### nodeDrain
+
+| Key | Description |
+| --- | --- |
+| timeOut | a time window to trigger the force node drain stretegy |
+| expectedNodeDrainTime | expected time in minutes for a single node drain to be finished, sed to setup the maintenance window |
+
+Example:
+```
+    nodeDrain:
+      timeOut: 45
+      expectedNodeDrainTime: 8
+```
+
+#### healthCheck
+
+| Key | Description |
+| --- | --- |
+| ignoredCriticals | a list of critical alerts need to be ignored in health check to unblock the upgrade process |
+| ignoredNamespaces | a list of namespaces need to be ignored in health check to unblock the upgrade process |
+
+Example:
+```
+    healthCheck:
+      ignoredCriticals:
+      - DNSErrors05MinSRE
+      - FluentdNodeDown
+      ignoredNamespaces:
+      - openshift-logging
+      - openshift-redhat-marketplace
+```
+
+#### extDependencyAvailabilityChecks
+
+| Key | Description |
+| --- | --- |
+| http | a list of external dependencies to check before the upgrade can be triggered |
+
+Example:
+```
+    extDependencyAvailabilityChecks:
+      http:
+        timeout: 10
+        urls:
+          - www.example.com
```

Comment:
Not actually sure what happens in MUO if a URL is specified without a leading protocol. (something to test)

```suggestion
          - https://www.example.com
```